lib/Scheduling/TestPasses.cpp

```diff
@@ -53,26 +63,39 @@ static LogicalResult constructProblem(Problem &prob, FuncOp func) {
     }
   }
 
-  // parse auxiliary dependences in the testcase, encoded as an array of
-  // 2-element arrays of integer attributes (see `test_asap.mlir`)
+  // parse auxiliary dependences in the testcase, encoded as an array of arrays
+  // of integer attributes
   if (auto attr = func->getAttrOfType<ArrayAttr>("auxdeps")) {
     auto &ops = prob.getOperations();
-    for (auto auxDepAttr : attr.getAsRange<ArrayAttr>()) {
-      if (auxDepAttr.size() != 2)
-        continue;
-      auto fromIdxAttr = auxDepAttr[0].dyn_cast<IntegerAttr>();
-      auto toIdxAttr = auxDepAttr[1].dyn_cast<IntegerAttr>();
-      if (!fromIdxAttr || !toIdxAttr)
-        continue;
-      unsigned fromIdx = fromIdxAttr.getInt();
-      unsigned toIdx = toIdxAttr.getInt();
-      if (fromIdx >= ops.size() || toIdx >= ops.size())
-        continue;
-
-      // finally, we have two integer indices in range of the operations list
-      if (failed(prob.insertDependence(
-              std::make_pair(ops[fromIdx], ops[toIdx])))) {
+    unsigned nOps = ops.size();
+    for (auto &elemArr : parseArrayOfArrays(attr)) {
+      if (elemArr.size() < 2 || elemArr[0] >= nOps || elemArr[1] >= nOps)
```

Comment:
Maybe turn this `if` into an `assert` (by negating the conditions)? Or maybe throw an error? Seems like if the invariant doesn't hold, we should complain to the compiler developer who wrote the test case.

Reply:
Good point, I changed it to `assert`.